flink-ml-core/src/main/java/org/apache/flink/ml/linalg/BLAS.java

```diff
@@ -32,9 +32,31 @@ public static double asum(DenseVector x) {
     }
 
     /** y += a * x . */
-    public static void axpy(double a, DenseVector x, DenseVector y) {
+    public static void axpy(double a, Vector x, DenseVector y) {
         Preconditions.checkArgument(x.size() == y.size(), "Vector size mismatched.");
-        JAVA_BLAS.daxpy(x.size(), a, x.values, 1, y.values, 1);
+        if (x instanceof SparseVector) {
+            axpy(a, (SparseVector) x, y);
+        } else {
+            axpy(a, (DenseVector) x, y);
+        }
+    }
+
+    /** Computes the hadamard product of the two vectors (y = y \hdot x). */
+    public static void hDot(Vector x, Vector y) {
```

Comment:
Good question! There has also been a discussion of the reason that BLAS does not provide a hadamard dot [1] --- Mostly saying that there are no performance gains.

However, I'd like to say that introducing `hDot` here is simply for convienice. I need to use it twice. 

By the way, we can replace the logic in MinMaxScaler if we have this method. This will certainly simplify the implementation.

What do you think?

[1] https://github.com/xianyi/OpenBLAS/issues/1083

Reply:
I think it would be a good idea to replace the logic in MinMaxScaler. But Weibo was concerned about the performance, as the implementation would be changed from something like 
```java
for (int i = 0; i < n; i++) {
  y[i] += a[i] * x[i];
}
```
to 
```java
for (int i = 0; i < n; i++) {
  x[i] *= a[i];
}
for (int i = 0; i < n; i++) {
  y[i] += x[i];
}
```
Thus I would like to hear from Weibo's opinion about this replacement.

Reply:
I talked with Weibo. He is fine with the change and also thinks it is better to keep the code consistent.